src/inspector_agent.cc

```diff
@@ -426,67 +426,53 @@ bool AgentImpl::OnInspectorHandshakeIO(inspector_socket_t* socket,
-    std::string str(b->base, read);
-    agent->PushPendingMessage(&agent->message_queue_, str);
-    free(b->base);
-
+    String16 str = String16::fromUTF8(buf->base, read);
+    free(buf->base);
+    PostIncomingMessage(str);
     // TODO(pfeldman): Instead of blocking execution while debugger
     // engages, node should wait for the run callback from the remote client
     // and initiate its startup. This is a change to node.cc that should be
     // upstreamed separately.
-    if (agent->wait_ && str.find("\"Runtime.run\"") != std::string::npos) {
-      agent->wait_ = false;
-      uv_sem_post(&agent->start_sem_);
+    if (wait_ && str.find("\"Runtime.run\"") != std::string::npos) {
+      wait_ = false;
+      uv_sem_post(&start_sem_);
     }
 
-    agent->platform_->CallOnForegroundThread(agent->parent_env_->isolate(),
-        new DispatchOnInspectorBackendTask(agent));
-    agent->parent_env_->isolate()
-        ->RequestInterrupt(InterruptCallback, agent);
-    uv_async_send(&agent->data_written_);
+    platform_->CallOnForegroundThread(parent_env_->isolate(),
+        new DispatchOnInspectorBackendTask(this));
+    parent_env_->isolate()->RequestInterrupt(InterruptCallback, this);
+    uv_async_send(&data_written_);
   } else if (read <= 0) {
     // EOF
-    if (agent->client_socket_ == socket) {
-      agent->client_socket_ = nullptr;
-      agent->platform_->CallOnForegroundThread(agent->parent_env_->isolate(),
-          new SetConnectedTask(agent, false));
-      uv_async_send(&agent->data_written_);
+    if (client_socket_ == socket) {
+      String16 message(TAG_DISCONNECT, sizeof(TAG_DISCONNECT) - 1);
+      client_socket_ = nullptr;
+      PostIncomingMessage(message);
+      uv_async_send(&data_written_);
     }
     DisconnectAndDisposeIO(socket);
   }
-  agent->pause_cond_.Broadcast(scoped_lock);
-}
-
-void AgentImpl::PushPendingMessage(std::vector<std::string>* queue,
-                                   const std::string& message) {
-  Mutex::ScopedLock scoped_lock(queue_lock_);
-  queue->push_back(message);
-}
-
-void AgentImpl::SwapBehindLock(std::vector<std::string> AgentImpl::*queue,
-                               std::vector<std::string>* output) {
-  Mutex::ScopedLock scoped_lock(queue_lock_);
-  (this->*queue).swap(*output);
+  pause_cond_.Broadcast(scoped_lock);
 }
 
 // static
 void AgentImpl::WriteCbIO(uv_async_t* async) {
   AgentImpl* agent = static_cast<AgentImpl*>(async->data);
   inspector_socket_t* socket = agent->client_socket_;
   if (socket) {
-    std::vector<std::string> outgoing_messages;
-    agent->SwapBehindLock(&AgentImpl::outgoing_message_queue_,
-                          &outgoing_messages);
-    for (auto const& message : outgoing_messages)
-      inspector_write(socket, message.c_str(), message.length());
+    MessageQueue outgoing_messages;
+    agent->SwapBehindLock(&agent->outgoing_message_queue_, &outgoing_messages);
+    for (const std::pair<int, String16>& m : outgoing_messages) {
```

Comment:
Switched to ::value_type - don't seem to be able to make cpplint work with const auto& - " Storage class (static, extern, typedef, etc) should be first.  [build/storage_class] [5]"